pkg/actuators/machine/actuator.go

```diff
@@ -116,9 +117,39 @@ func (a *Actuator) Create(context context.Context, cluster *clusterv1.Cluster, m
 		}
 		return err
 	}
+	err = a.updateProviderID(machine, instance)
+	if err != nil {
+		return fmt.Errorf("failed to update machine object with providerID: %v", err)
+	}
 	return a.updateStatus(machine, instance)
```

Comment:
If we just updated the providerID isn't `machine` on this line a different version of the object now?